controllers/managedocs_controller.go

```diff
@@ -2045,3 +2096,28 @@ func (r *ManagedOCSReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDeviceS
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *ManagedOCSReconciler) reconcileODFSubscription() error {
+	subList := &opv1a1.SubscriptionList{}
+	if err := r.list(subList); err != nil {
+		return fmt.Errorf("Unable to list subscriptions %v", err)
+	}
+	actualSub := &opv1a1.Subscription{}
+	for index := range subList.Items {
+		item := subList.Items[index]
+		if item.Spec.Package == odfPackageName {
+			actualSub = &item
+		}
+	}
+	if actualSub.Spec.Channel == desiredODFSubscriptionChannel {
+		r.Log.Info("Current ODF subscription channel is same as desired ODF subscription channel, not updating")
+		return nil
+	}
+	r.Log.Info("Updating ODF subsctiption channel", "current channel", actualSub.Spec.Channel, "desired channel", desiredODFSubscriptionChannel)
+	actualSub.Spec.Channel = desiredODFSubscriptionChannel
+
+	if err := r.update(actualSub); err != nil {
+		return fmt.Errorf("Unable to update %s subscription %v", odfPackageName, err)
```

Comment:
You are not "unable", you "failed"!
```suggestion
		return fmt.Errorf("Failed to update %s subscription %v", odfPackageName, err)
```

Reply:
Updated